sdc/str_arr_ext.py

```diff
@@ -670,33 +670,43 @@ def construct_string_array(context, builder):
 @lower_builtin(StringArray, types.UniTuple)
 @lower_builtin(StringArray, types.Tuple)
 def impl_string_array_single(context, builder, sig, args):
-    if isinstance(args[0], types.UniTuple):
-        assert args[0].dtype == string_type
 
-    if isinstance(args[0], types.Tuple):
-        for i in args[0]:
+    arg = args[0]
+    if isinstance(arg, (types.UniTuple, types.List)):
+        assert (arg.dtype == string_type
+                or (isinstance(arg.dtype, types.optional) and arg.type == string_type))
```

Comment:
Shouldn't `arg.type == string_type` be replaced with `arg.dtype.type == string_type` and `types.optional` with `types.Optional` as above?